src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,57 @@
+import { forwardRef } from 'react';
+import classnames from 'classnames';
+import { getCommonProps, px } from '../../utils';
+import { getScssVar } from '../../utils/scssUtils';
+import * as iconComponents from '../../assets/formatted';
+
+export type IconName = keyof typeof iconComponents;
+
+export interface IconProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Height of the icon in px (number) or as a string with units (e.g. '1em')
+   */
+  height?: number | string | null;
+  /**
+   * Width of the icon in px (number) or as a string with units (e.g. '1em')
+   */
+  width?: number | string | null;
+  /**
+   * Color of the icon. Only accepts valid seldon color tokens. Defaults to $pure-black
+   */
+  color?: string;
+  /**
+   * Name of Icon to render
+   */
+  icon: IconName;
+}
+
+/**
+ * ## Overview
+ *
+ * Component for rendering icons. Accepts an icon name and renders the corresponding SVG icon at the height and width specified. Also accepts a color parameter to change the fill color of the icon, but not all icons can be recolored (Business logos for example).
```

Comment:
are we ever gonna want to be able to specify fill vs. stroke?

Reply:
Hmm, that's a possibility, but it might get confusing since some svgs won't support custom colors (for example, we have a spotify logo in there that doesn't use the color prop). Ultimately, this step is manual and it'll have to be up to the dev and the reviewers to make sure the fill or stroke is set properly